### DISCUSSION TITLE
I am curious as to why the assert() function is used.

### DISCUSSION THREAD

**Discussion by dhkstnrh** [OpenDDS/OpenDDS#4289] [Q&A] [answered]
Hello!

I am a student studying DDS. I have two questions while studying.

First, as far as I know, I don't compile the Assert() function through #defineNDEBUG before I deploy it in the release version, Within that DDS code, the Assert() function appears to be Compile.

Is there any particular reason why you didn't remove this function from the developer's point of view? Can you explain it in detail if there is a special reason??
If there's no specific reason, why didn't you remove it?

Second, when you deploy that DDS for commercial use, I wonder if you remove all of the Assert() functions and deploy them!

**Comment by jrw972** [comment, marked as answer]
`OPENDDS_ASSERT` is defined in `dds/DCPS/Definitions.h`.  It is defined in terms of `assert` so defining `NDEBUG` should cause it to become a no-op.  If the assert code is still present, you should double-check your build configuration.

Using `assert` is based on the concept of invariants or things that must always be true.  If the condition expressed by an invariant is not true, then it means that there is a bug in the code, i.e., the invariant is not an invariant.  Using `assert` in this way allows us to document what the invariants are and find bugs.  Invariants should not be used for error checking or handling as run-time errors can and do happen.
